scripts/release.py

```diff
@@ -0,0 +1,199 @@
+                continue
+            if patch_num > patch_version:
+                patch_version = patch_num
+
+        name = "%s.%s" % (base, str(patch_version + 1))
+        tag = "v%s" % name
+        rn_raw = generate_rn(base)
+        rn = clean_rn(rn_raw)
+        create_draft_release(branch=base, name=name, tag=tag, dd_repo=dd_repo, rn=rn, prerelease=False)
+
+    # minor release
+    else:
+        name = "%s.0" % base
+        tag = "v%s" % name
+        branch = base
+        rn_raw = generate_rn(branch)
+        # get anything in unreleased section in case there were updates since the last RC
+        unreleased = (
+            rn_raw.decode().split("## v")[0].replace("\n## Unreleased\n", "", 1).replace("# Release Notes\n", "", 1)
+        )
+        unreleased = unreleased.split("###")
+        try:
+            unreleased_sections = dict(section.split("\n\n-") for section in unreleased)
+        except ValueError:
+            unreleased_sections = {}
+        relevant_rns = []
+        if unreleased_sections:
+            relevant_rns.append(unreleased_sections)
+
+        rns = rn_raw.decode().split("## v")
+        split_str = "## v%s" % base
+        for rn in rns:
+            if rn.startswith("%s.0" % base):
+                # cut out the version section
+                sections = rn.split("###")[1:]
+                prelude_section = {}
+                # if there is a prelude, we need to grab that separately since it has different syntax
+                if sections[0].startswith(" Prelude\n\n"):
+                    prelude_section[" Prelude"] = sections[0].split("\n\n", 1)[1]
+                    sections = sections[1:]
+                sections_dict = {**dict(section.split("\n\n", 1) for section in sections), **prelude_section}
+                relevant_rns.append(sections_dict)
+        # join all the sections from different relevant RCs together
+        keys = set().union(*relevant_rns)
+        rns_dict = {k: "".join(dic.get(k, "") for dic in relevant_rns) for k in keys}
+        rns_dict_clean = {}
+        for key in rns_dict.keys():
+            rns_dict_clean[key.lstrip()] = rns_dict[key]
+
+        # combine the release note sections into a string in the correct order
+        rn_clean = ""
+        rn_key_order = [
+            "Prelude",
+            "New Features",
+            "Known Issues",
+            "Upgrade Notes",
+            "Deprecation Notes",
+            "Bug Fixes",
+            "Other Changes",
+        ]
+        for key in rn_key_order:
+            try:
+                rn_clean += "### %s\n\n%s" % (key, rns_dict_clean[key])
+            except KeyError:
+                continue
+
+        create_draft_release(branch=branch, name=name, tag=tag, dd_repo=dd_repo, rn=rn_clean, prerelease=False)
+
+
+def clean_rn(rn_raw):
+    return rn_raw.decode().split("## v")[0].replace("\n## Unreleased\n", "", 1).replace("# Release Notes\n", "", 1)
```

Comment:
Does this remove all release notes generated other than the one of interest?

Reply:
Yes exactly, I'll comment it.